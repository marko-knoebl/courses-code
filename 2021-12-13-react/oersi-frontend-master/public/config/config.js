window['runTimeConfig'] = {
  ELASTIC_SEARCH: {
    URL: "http://localhost:9200/",
    CREDENTIALS: "",
    APP_NAME: "oer_data"
  },
  GENERAL_CONFIGURATION:{
    PUBLIC_URL: "http://localhost/resources",
    RESULT_PAGE_SIZE_OPTIONS:["12", "24", "48", "96"],  // page size options configuration    
    NR_OF_RESULT_PER_PAGE:12,  //  number of results to show per view. Defaults to 12.
    /**
     * Accept a list of objects 
     * example:
     * {'path': 'public/{folderName}/{languageCode}/{fileName}.html', 'language': '{languageCode}'}
     * 
     */
    PRIVACY_POLICY_LINK: [],
    I18N_CACHE_EXPIRATION: 600000, // expiration time of the i18n translation cache storage
    I18N_DEBUG: false,
    TRACK_TOTAL_HITS: true, // track number of total hits from elasticsearch - see https://www.elastic.co/guide/en/elasticsearch/reference/7.10/search-your-data.html#track-total-hits
    FEATURES: {
      EMBED_OER: true, // feature toggle: use "embed-oer" button
      SCROLL_TOP_BUTTON: true, // feature toggle: use "scroll-to-top" button
      USE_RESOURCE_PAGE: true // feature toggle: use a single html-page per resource
    }
  }
}