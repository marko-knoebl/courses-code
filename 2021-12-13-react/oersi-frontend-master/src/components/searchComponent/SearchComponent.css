.search-component {
  width: 40%;
}

svg.search-icon {
  fill: #e86161 !important;
  height: 1.8em;
}

/* Responsive styling */
@media (max-width: 991.98px) {
  .search-component {
    margin-bottom: 10px;
    width: 100%;
  }
}
