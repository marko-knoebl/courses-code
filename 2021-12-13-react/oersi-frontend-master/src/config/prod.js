export default {
  name: "production",
  resultList: {
    component: "results",
    dataField: "name",
    pagination: true,
    showResultStats: true,
    paginationAt: "bottom",
    sortBy: null,
    pagesShow: 5,
    sizeShow: 5,
    URLParams: true,
    showEndPage: true,
    and: [
      "author",
      "license",
      "search",
      "provider",
      "learningResourceType",
      "language",
      "about",
      "sourceOrganization",
    ],
    sortByDynamic: null,
  },
  searchComponent: {
    component: "search",
    dataField: ["name", "creator.name", "description", "keywords"],
    fieldWeights: [1, 3],
    queryFormat: "and",
    fuzziness: 0,
    debounce: 100,
    autosuggest: true,
    highlight: true,
    highlightField: "keywords",
    iconPosition: "right",
    showFilter: true,
    URLParams: true,
    and: [
      "author",
      "license",
      "provider",
      "results",
      "learningResourceType",
      "language",
      "about",
      "sourceOrganization",
    ],
  },
  multiList: [
    {
      component: "about",
      dataField: "about.id",
      title: "about",
      placeholder: "about",
      filterLabel: "about",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      size: 100,
      className: "about-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "provider",
        "results",
        "language",
        "learningResourceType",
        "sourceOrganization",
      ],
    },
    {
      component: "learningResourceType",
      dataField: "learningResourceType.id",
      title: "resourceType",
      placeholder: "resourceType",
      filterLabel: "resourceType",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      className: "lrt-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "provider",
        "results",
        "language",
        "sourceOrganization",
        "about",
      ],
    },
    {
      component: "license",
      dataField: "license.id",
      title: "license",
      placeholder: "license",
      filterLabel: "license",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      className: "license-card",
      fontAwesome: "",
      URLParams: true,
      customQuery: (value, props) => {
        return value instanceof Array
          ? {
              query: {
                bool: {
                  should: [
                    ...value.map((v) => ({
                      prefix: {
                        "license.id": v,
                      },
                    })),
                    ...value.map((v) => ({
                      prefix: {
                        "license.id": v.replace("https:/", "http:/"),
                      },
                    })),
                  ],
                },
              },
            }
          : {}
      },
      defaultQuery: getPrefixAggregationQuery("license.id", [
        "https://creativecommons.org/licenses/by/",
        "https://creativecommons.org/licenses/by-sa/",
        "https://creativecommons.org/licenses/by-nd/",
        "https://creativecommons.org/licenses/by-nc-sa/",
        "https://creativecommons.org/licenses/by-nc/",
        "https://creativecommons.org/licenses/by-nc-nd/",
        "https://creativecommons.org/publicdomain/zero/",
        "https://creativecommons.org/publicdomain/mark",
      ]),
      and: [
        "author",
        "search",
        "provider",
        "results",
        "learningResourceType",
        "language",
        "about",
        "sourceOrganization",
      ],
    },
    {
      component: "author",
      dataField: "persons.name.keyword",
      nestedField: "",
      title: "author",
      placeholder: "author",
      filterLabel: "author",
      queryFormat: "and",
      showMissing: true,
      showSearch: true,
      showFilter: true,
      size: 1000,
      className: "author-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "license",
        "search",
        "provider",
        "results",
        "learningResourceType",
        "language",
        "about",
        "sourceOrganization",
      ],
    },
    {
      component: "sourceOrganization",
      dataField: "institutions.name",
      title: "organization",
      placeholder: "organization",
      filterLabel: "organization",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: true,
      size: 100,
      className: "source-type-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "provider",
        "results",
        "language",
        "learningResourceType",
        "about",
      ],
    },
    {
      component: "language",
      dataField: "inLanguage",
      title: "language",
      placeholder: "language",
      filterLabel: "language",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      className: "language-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "provider",
        "results",
        "learningResourceType",
        "about",
        "sourceOrganization",
      ],
    },
    {
      component: "provider",
      dataField: "mainEntityOfPage.provider.name",
      title: "provider",
      placeholder: "provider",
      filterLabel: "provider",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      className: "provider-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "results",
        "learningResourceType",
        "language",
        "about",
        "sourceOrganization",
      ],
    },
  ],
}
function getPrefixAggregationQuery(fieldName, prefixList) {
  var aggsScript = "if (doc['" + fieldName + "'].size()==0) { return null }"
  aggsScript += prefixList.reduce(
    (result, prefix) =>
      result +
      " else if (doc['" +
      fieldName +
      "'].value.startsWith('" +
      prefix +
      "') || doc['" +
      fieldName +
      "'].value.startsWith('" +
      prefix.replace("https:/", "http:/") +
      "')) { return '" +
      prefix +
      "'}",
    ""
  )
  aggsScript += " else { return doc['" + fieldName + "'] }"
  return () => ({
    aggs: {
      "license.id": {
        terms: {
          script: {
            source: aggsScript,
            lang: "painless",
          },
        },
      },
    },
  })
}
