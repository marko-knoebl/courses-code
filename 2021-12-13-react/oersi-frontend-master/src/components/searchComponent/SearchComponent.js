import React, {useState} from "react"
import {DataSearch} from "@appbaseio/reactivesearch"
import {useHistory, useLocation} from "react-router-dom"
import "./SearchComponent.css"
import config from "react-global-configuration"
import PropTypes from "prop-types"
import {withTranslation} from "react-i18next"
/**
 * Search Component
 * creates a search box UI component that is connected to one or more database fields,
 * and you ca define those fields from src/config/prod.json
 *
 * @author Edmond Kacaj <edmondikacaj@gmail.com>
 * @props Properties from Parent Component
 */
const SearchComponent = (props) => {
  //declare varibale to get data from Configuration fle prod.json
  const [conf] = useState(config.get("searchComponent"))
  const history = useHistory()
  const location = useLocation()

  const redirect = (value) => {
    if (location.pathname !== "/") {
      const search = value ? `?${conf.component}="${value}"` : null
      history.replace({pathname: location.pathname, search: search}) // replace current entry to be able to move back
      history.push({pathname: "/", search: search})
    }
  }
  return (
    <div className="search-component">
      <DataSearch
        componentId={conf.component}
        placeholder={props.t("SEARCH_COMPONENT.PLACEHOLDER")}
        dataField={conf.dataField}
        fieldWeights={conf.fieldWeights}
        queryFormat={conf.queryFormat}
        fuzziness={conf.fuzziness}
        debounce={conf.debounce}
        autosuggest={conf.autosuggest}
        highlight={conf.highlight}
        highlightField={conf.highlightField}
        customHighlight={() => ({
          highlight: {
            pre_tags: ["<mark>"],
            post_tags: ["</mark>"],
            fields: {
              text: {},
              title: {},
            },
            number_of_fragments: 0,
          },
        })}
        innerClass={{
          title: "search-title",
          input: "search-component-input",
          mic: "search-component-img",
        }}
        searchInputId="NameSearch"
        iconPosition={conf.iconPosition}
        showFilter={conf.showFilter}
        URLParams={conf.URLParams}
        react={{
          and: conf.and,
        }}
        renderNoSuggestion={() => onNoSuggestion()}
        renderError={(error) => onError(error)}
        onValueSelected={redirect}
      />
    </div>
  )
  function onError(error) {
    return (
      <div>
        Something went wrong!
        <br />
        Error details
        <br />
        {error}
      </div>
    )
  }

  function onNoSuggestion() {
    return <div>No suggestions found</div>
  }
}

SearchComponent.propTypes = {
  conf: PropTypes.object,
}

export default withTranslation()(SearchComponent)
