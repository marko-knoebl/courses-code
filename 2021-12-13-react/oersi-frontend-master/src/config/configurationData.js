import config from "react-global-configuration"
import prod from "./prod"

export const registerConfiguration = () => {
  config.set(prod, {freeze: false})
}
