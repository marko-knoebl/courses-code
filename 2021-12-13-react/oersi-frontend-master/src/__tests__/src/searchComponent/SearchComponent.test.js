import React from "react"
import {shallow} from "../../../setupTests"
import {registerConfiguration} from "../../../config/configurationData"
import SearchComponent from "../../../components/searchComponent/SearchComponent"

describe("SearchComponent ==> Test UI  ", () => {
  registerConfiguration()
  const wrapperShadow = shallow(<SearchComponent />)
  it("SearchComponent : should render correctly", () => {
    expect(wrapperShadow).toMatchSnapshot()
  })
})
