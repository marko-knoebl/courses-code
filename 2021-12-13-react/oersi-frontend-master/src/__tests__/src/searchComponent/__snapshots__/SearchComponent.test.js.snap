// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent ==> Test UI   SearchComponent : should render correctly 1`] = `ShallowWrapper {}`;
